main.go

```diff
@@ -137,25 +156,24 @@ func caduceus(arguments []string) int {
 		return 1
 	}
 
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create %v metric: %v \n", IncomingQueueLatencyHistogram, err)
+		return 1
+	}
+
 	serverWrapper := &ServerHandler{
 		Logger: logger,
 		caduceusHandler: &CaduceusHandler{
 			senderWrapper: caduceusSenderWrapper,
 			Logger:        logger,
 		},
-		errorRequests:            metricsRegistry.NewCounter(ErrorRequestBodyCounter),
-		emptyRequests:            metricsRegistry.NewCounter(EmptyRequestBodyCounter),
-		invalidCount:             metricsRegistry.NewCounter(DropsDueToInvalidPayload),
-		incomingQueueDepthMetric: metricsRegistry.NewGauge(IncomingQueueDepth),
-		modifiedWRPCount:         metricsRegistry.NewCounter(ModifiedWRPCounter),
-		maxOutstanding:           0,
-		// 0 is for the unused `buckets` argument in xmetrics.Registry.NewHistogram
-		incomingQueueLatency: metricsRegistry.NewHistogram(IncomingQueueLatencyHistogram, 0),
-		now:                  time.Now,
+		metrics:        serverHandlerMetrics,
```

Comment:
updated `ServerHandler` to simplify and prep for the fx integration by introducing `ServerHandlerMetrics`